core/src/main/java/com/graphhopper/routing/DijkstraBidirectionCH.java

```diff
@@ -62,10 +64,16 @@ private boolean entryIsStallable(SPTEntry entry, IntObjectMap<SPTEntry> bestWeig
         // reached via a suboptimal path. We do this regardless of the CH level of the adjacent nodes.
         EdgeIterator iter = edgeExplorer.setBaseNode(entry.adjNode);
         while (iter.next()) {
+            // no need to inspect the edge we are coming from
+            if (iter.getEdge() == entry.adjNode) {
```

Comment:
First I was hoping this might improve performance, but not really. In the `iter.getEdge() == entry.adjNode` case `adjNode.weight+weighting.calcWeight` always exactly equals `entry.weight` below and in this case we never want to stall.

Reply:
Oh this was a stupid error, it should be `entry.edge` instead of `entry.adjNode`, fixed in ec01e5e458bce46ffddea8a7b266cacbc49e6a1e